jax/_src/numpy/polynomials/chebyshev.py

```diff
@@ -0,0 +1,605 @@
+    c1, c2 = _promote_dtypes_inexact(c1, c2)
+    if len(c1) > len(c2):
+        c1 = c1.at[:c2.size].add(-c2)
+        ret = c1
+    else:
+        c2 = -c2
+        c2 = c2.at[:c1.size].add(c1)
+        ret = c2
+    return ret
+
+@jit
+def chebmulx(c):
+    [c] = pu.as_series([c])
+    c = _promote_dtypes_inexact(c)[0]  #Surely there is a better way to do this
+    prd = jnp.empty(len(c) + 1, dtype=c.dtype)
+    prd = prd.at[0].set(c[0]*0)
+    prd = prd.at[1].set(c[0])
+    tmp = c[1:]/2
+    prd = prd.at[2:].set(tmp)
+    prd = prd.at[0:-2].add(tmp)
+    return prd
+
+@partial(jit, static_argnames=('trim',))
+def chebmul(c1, c2, trim=False):
+    [c1, c2] = pu.as_series([c1, c2])
+    c1, c2 = _promote_dtypes_inexact(c1, c2)
+    z1 = _cseries_to_zseries(c1)
+    z2 = _cseries_to_zseries(c2)
+    prd = _zseries_mul(z1, z2)
+    ret = _zseries_to_cseries(prd)
+    if trim:
+        return pu.trimseq(ret)  #NO JIT
+    else:
+        return ret
+
+@partial(jit, static_argnames=('trim',))
+def chebdiv(c1, c2, trim=False):
+    [c1, c2] = pu.as_series([c1, c2])
+    c1, c2 = _promote_dtypes_inexact(c1, c2)
+    lc1 = len(c1)
+    lc2 = len(c2)
+    if lc1 < lc2:
+        return c1.at[:1].multiply(0), c1
+    elif lc2 == 1:
+        return c1/c2[-1], c1.at[:1].multiply(0)
+    else:
+        z1 = _cseries_to_zseries(c1)
+        z2 = _cseries_to_zseries(c2)
+        quo, rem = _zseries_div(z1, z2)
+        quo = _zseries_to_cseries(quo)
+        rem = _zseries_to_cseries(rem)
+        if trim:
+            quo = pu.trimseq(quo)  #NO JIT
+            rem = pu.trimseq(rem)  #NO JIT
+        return quo, rem
+
+@partial(jit, static_argnames=('pow', 'maxpower'))
+def chebpow(c, pow, maxpower=16):
+    [c] = pu.as_series([c])
+    c = _promote_dtypes_inexact(c)[0]  #Surely there is a better way to do this
+    power = int(pow)
+    if power != pow or power < 0:
+        raise ValueError("Power must be a non-negative integer.")
+    elif maxpower is not None and power > maxpower:
+        raise ValueError("Power is too large.")
+    elif power == 0:
+        return jnp.array([1], dtype=c.dtype)
+    elif power == 1:
+        return c
+    else:
+        zs = _cseries_to_zseries(c)
+        prd = zs
+        for i in range(2, power+1):
+            prd = jnp.convolve(prd, zs)
+        return _zseries_to_cseries(prd)
+
+@partial(jit, static_argnames=('m', 'axis'))
+def chebder(c, m=1, scl=1, axis=0):
+    c = jnp.array(c, ndmin=1, copy=True)
+    c = _promote_dtypes_inexact(c)[0]  #Surely there is a better way to do this
+    cnt = int(m)
+    iaxis = int(axis)
+    if cnt < 0:
+        raise ValueError("The order of derivation must be non-negative")
+
+    if cnt == 0:
+        return c
+
+    c = jnp.moveaxis(c, iaxis, 0)
+    n = len(c)
+    if cnt >= n:
+        c = c.at[:1].multiply(0)
+        c = c[:1]
+    else:
+        for i in range(cnt):
+            n = n-1
+            c *= scl
+            der = jnp.empty((n,) + c.shape[1:], dtype=c.dtype)
```

Comment:
should this be `jnp.zero` rather than `jnp.empty`? `empty` implies uninitialized values, and below it looks like you're adding numbers to them.